db/updateItem.go

```diff
@@ -39,8 +41,46 @@ func (d *dynamoDB) UpdateItem(listID string, itemID string, newName string) (*da
 		if e.Code() == dynamodb.ErrCodeConditionalCheckFailedException {
 			return nil, ErrorNotFound
 		}
+		if e.Code() == "ValidationException" { // https://github.com/aws/aws-sdk-go/issues/3140
+			return nil, ErrorBadRequest
+		}
+		return nil, err
 	default:
 		return nil, err
 	}
-	return item, nil
+
+	item := new(data.Item)
+	err = dynamodbattribute.UnmarshalMap(output.Attributes, &item)
```

Comment:
Because we specified `ReturnValues == "ALL_NEW"` the updated item is returned in the output